infra/cache/kInfraRedisCacheWrapper.php

```diff
@@ -0,0 +1,314 @@
+<?php
+
+require_once(dirname(__FILE__) . '/kInfraBaseCacheWrapper.php');
+
+/**
+ * @package infra
+ * @subpackage cache
+ */
+class kInfraRedisCacheWrapper extends kInfraBaseCacheWrapper
+{
+	protected $hostName;
+	protected $port;
+	protected $timeout;
+	protected $statsKey;
+
+	protected $redis = null;
+	protected $gotError = false;
+	protected static $_stats = array();
+	protected $connectAttempts = 0;
+	protected $persistent = false;
+	protected $cluster = false;
+
+	const STAT_CONN = 'conn';
+	const STAT_OP = 'op';
+	const STAT_COUNT = 'count';
+	const STAT_TIME = 'time';
+	const MAX_CONNECT_ATTEMPTS = 4;
+
+	/* (non-PHPdoc)
+     * @see kBaseCacheWrapper::doInit()
+     */
+	protected function doInit ($config)
+	{
+		if (!class_exists('Redis'))
+		{
+			self::safeLog("Redis class doesn't exists, can't connect without it");
+			return false;
+		}
+
+		if (!isset($config['host']) || !isset($config['port']))
+		{
+			self::safeLog("Missing host or port in config, can't connect without it");
+			return false;
+		}
+
+		$this->hostName = $config['host'];
+		$this->port = $config['port'];
+		$this->timeout = $config['timeout'];
+		if (isset($config['persistent']) && $config['persistent'])
+			$this->persistent = true;
+		if (isset($config['cluster']) && $config['cluster'])
+			$this->cluster = true;
+
+		return $this->reconnect();
+	}
+
+	public function __destruct()
+	{
+		$this->close();
+	}
+
+	protected function close()
+	{
+		if ($this->redis)
+		{
+			$this->redis->close();
+		}
+		$this->redis = null;
+	}
+
+	/* (non-PHPdoc)
+     * @see kBaseCacheWrapper::get()
+     */
+	protected function doGet ($key)
+	{
+		return $this->callAndDetectErrors('get',array($key));
```

Comment:
Add spacing after the , 
For example
$this->callAndDetectErrors('get', array($key));